### PR TITLE
Roll dart, zlib, emscripten, tweak license script for rolled zlib

### PR DESCRIPTION
**Pull request by aam**
Changes since last dart roll
```
1f224df52be Version 3.1.0-15.0.dev
3c284a0c95c Revert "Reland "[VM] Begin supporting Perfetto file recorder""
754aca00499 Bump github/codeql-action from 2.2.11 to 2.2.12
7c8e61104f6 Bump actions/checkout from 3.5.0 to 3.5.2
f9f1030959d (origin/base) [infra] DEPS and test dart-lang/native packages
b04b0844067 [vm/win7] Dynamically load functions for unwinding instructions.
a6ebd6fa8e9 Bump tools to 545d7e1c73ce21b8c91f638021f9d487d324a501
c4795a75508 [vm, compiler] Don't do safepoint transitions in generated code under TSAN.
2fcf9754032 Version 3.1.0-14.0.dev
393b1016292 Version 3.1.0-13.0.dev
39fc15a0b4c (origin/lkgr) [analysis_server] Add missing declaration semantic token modifier on classes/constructors
609db33dc7a Issue 51985. Quick fix to add missing pattern case to switch expression/pattern.
8fa08ae80b6 Add 'Convert to if-case statement' quick assist.
17d92f90d0b offer wildcard suggestions for destructured record fields
3149f813bbf [dartdev] Update --enable-analytics to re-enable unified analytics
eee509140dd Add dependency overrides for pkg/analyzer
3069c4f3e2b [analysis_server] Add missing "constructor" semantic token modifiers on constructor declarations
9834f87c542 [dart2js] Fix noSuchMethod override handling.
639a6f06ef9 Simplify several non-terminals to use regex-like grammar operators
f88e3bd77c5 Issue 51903. Test fixes for ParserErrorCode.VAR_AND_TYPE
f4d42303ad2 [analysis_server] Handle completion in whitespace after string values
70762f0de0e [ANTLR] Transform recursive definition of onParts to use iteration ...
453647022f6 [analysis_server] Don't fail on invalid/complex setters in LSP code completion
a565638b910 Issue 51689. Report when 'super' is used as a standalone expression, not as a receiver of an invocation.
b78d89ce0c7 Fix DynamicTypeImpl to InterfaceTypeImpl cast exception in AddTypeAnnotation, when ambiguous SetOrMapLiteral.
bd9c7a46e2b Roll zlib to 14dd4c4455602c9b71a1a89b5cafd1f4030d2e3f
18b5ecdfc99 [vm] Cleanup dead code around TypeArgumentClassFinder
9d4d48949c0 Add a top-level variable strict-inference test case
8df1b888771 [VM/Service] Create JSONBase64String class
b4aa83b0b7f [VM/Service] Rename JSONStream::AppendSerializedObject to JSONStream::AppendBytes
4cd9c9c6666 Reland "[VM] Begin supporting Perfetto file recorder"
fa3a72fa7e5 [cfe/ffi] Error on compound constructors
```

### CODE REVIEW

**Comment by aam** [comment]
@Hixie fyi - rolling of the zlib requires some tweaking to the license script: .3 is an exetnsion of man-page doc file plus some other minor tweks.

File: tools/licenses/lib/filesystem.dart [hunk 1]
```diff
@@ -257,6 +257,7 @@ FileType identifyFile(String name, Reader reader) {
     case '.diff':
       // Don't try to read the copyright out of patch files, since there'll be fragments.
       return FileType.binary;
+    case '.3':
```
Comment (Hixie) [inline tools/licenses/lib/filesystem.dart:260]: i think we do .1 in the `paths.dart` file, fwiw. might be worth making the two consistent with each other.

Comment (Hixie) [reply tools/licenses/lib/filesystem.dart:260]: skipping entirely (i.e. doing it in `paths.dart`) may be more correct if we don't ship the man file (which i assume we don't).

Comment (aam) [reply tools/licenses/lib/filesystem.dart]: Ah, interesting. Thanks, done.

**Review by zanderso — APPROVED** [review]
LGTM

**Review by eyebrowsoffire — APPROVED** [review]

**Pull request merged**